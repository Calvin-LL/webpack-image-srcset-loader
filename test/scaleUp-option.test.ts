import webpack from "webpack";

import compile from "./helpers/compile";
import execute from "./helpers/execute";
import getCompiler from "./helpers/getCompiler";
import readAsset from "./helpers/readAsset";

describe.each([4, 5] as const)('v%d "scaleUp" option', (webpackVersion) => {
  test("should not skip original size", async () => {
    const compiler = getCompiler(webpackVersion, {
      sizes: ["2912w"],
    });
    const stats = await compile(webpackVersion, compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  test("should skip greater size", async () => {
    const compiler = getCompiler(webpackVersion, {
      sizes: ["2913w", "1x"],
    });
    const stats = await compile(webpackVersion, compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });

  test("should work with true", async () => {
    const compiler = getCompiler(webpackVersion, {
      sizes: ["2913w"],
      scaleUp: true,
    });
    const stats = await compile(webpackVersion, compiler);

    expect(
      execute(readAsset("main.bundle.js", compiler, stats as webpack.Stats))
    ).toMatchSnapshot("result");
  });
});
