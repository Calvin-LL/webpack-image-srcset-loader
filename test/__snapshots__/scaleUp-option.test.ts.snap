// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 "scaleUp" option should not skip original size: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2912w"`;

exports[`v4 "scaleUp" option should skip greater size: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 1x"`;

exports[`v4 "scaleUp" option should work with true: result 1`] = `"a5e4cd08cdc5d90da660e7acfb548473.jpg 2913w"`;

exports[`v5 "scaleUp" option should not skip original size: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 2912w"`;

exports[`v5 "scaleUp" option should skip greater size: result 1`] = `"fdaf70a107ad3cebfaf1e03ff0601c1c.jpg 1x"`;

exports[`v5 "scaleUp" option should work with true: result 1`] = `"a5e4cd08cdc5d90da660e7acfb548473.jpg 2913w"`;
